ledger/sandbox/src/main/scala/com/digitalasset/platform/apiserver/TimeServiceBackendClock.scala

```diff
@@ -0,0 +1,17 @@
+// Copyright (c) 2020 The DAML Authors. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package com.digitalasset.platform.apiserver
+
+import java.time.{Clock, Instant, ZoneId, ZoneOffset}
+
+class TimeServiceBackendClock(
```

Comment:
We already had this and removed it. Would a `() => Instant` do?

Reply:
Well, as you've asked twice… 😁

Reply:
I edited the previous comment when I noticed this. In general I like the idea of providing clocks as a dependency to classes that need to access time, makes testing easier. The lambda is simply easier and doesn't have the concept of timezone, which doesn't apply to us. I commented mostly because I was the one who introduced a clock in the past and then got convinced to remove it afterwards.